dask_sql/physical/rel/logical/table_scan.py

```diff
@@ -48,11 +51,16 @@ def convert(
         df = dc.df
         cc = dc.column_container
 
-        # Make sure we only return the requested columns
-        row_type = table.getRowType()
-        field_specifications = [str(f) for f in row_type.getFieldNames()]
-        cc = cc.limit_to(field_specifications)
+        # If the 'TableScan' instance contains projected columns only retrieve those columns
+        # otherwise get all projected columns from the 'Projection' instance, which is contained
+        # in the 'RelDataType' instance, aka 'row_type'
+        if table_scan.containsProjections():
```

Comment:
Not directly related to invalid projections, but we based on this, it needs to be seen if predicate pushdown is working as expected. 
Probably a followup issue to investigate this.

Reply:
I tested locally and this does enable that it is working. I will add a pytest proving that however.

Reply:
@ayushdg just added the test, let me know what you think. I read a locally generated Parquet file with differing combinations of columns and validate just the specified columns are read.